Core/include/Acts/TrackFitting/detail/GsfActor.hpp

```diff
@@ -604,29 +610,47 @@ struct GsfActor {
         is_valid_measurement = true;
       }
 
-      result.currentTips.push_back(trackStateProxy.index());
-      result.weightsOfStates[result.currentTips.back()] = cmp.weight();
+      tmpStates.tips.push_back(trackStateProxy.index());
+      tmpStates.weights[tmpStates.tips.back()] = cmp.weight();
     }
 
-    computePosteriorWeights(*result.fittedStates, result.currentTips,
-                            result.weightsOfStates);
+    computePosteriorWeights(tmpStates.traj, tmpStates.tips, tmpStates.weights);
 
-    detail::normalizeWeights(result.currentTips, [&](auto idx) -> double& {
-      return result.weightsOfStates.at(idx);
+    detail::normalizeWeights(tmpStates.tips, [&](auto idx) -> double& {
+      return tmpStates.weights.at(idx);
     });
 
     // Do the statistics
     ++result.processedStates;
 
-    // We also need to save outlier states here, otherwise they would not be
-    // included in the MT if they are at the end of the track
-    result.lastMeasurementTips = result.currentTips;
-
     // TODO should outlier states also be counted here?
     if (is_valid_measurement) {
       ++result.measurementStates;
     }
 
+    addCombinedState(result, tmpStates, surface);
+    result.lastMeasurementTip = result.currentTip;
+
+    using FiltProjector =
+        MultiTrajectoryProjector<StatesType::eFiltered, traj_t>;
+    FiltProjector proj{tmpStates.traj, tmpStates.weights};
+
+    std::vector<std::tuple<double, BoundVector, BoundMatrix>> v;
+
+    // TODO why can 0 weight happen?
```

Comment:
does it still?

Reply:
yes, unfortunately. But I would adress this later